lib/mongo/crypt/handle.rb

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'ffi'
+require 'base64'
+
+module Mongo
+  module Crypt
+
+    # A handle to the libmongocrypt library that wraps a mongocrypt_t object,
+    # allowing clients to set options on that object or perform operations such
+    # as encryption and decryption
+    class Handle
+
+      # Creates a new Handle object and initializes it with options
+      #
+      # @example Instantiate a Handle object with local KMS provider options
+      #   Mongo::Crypt::Handle.new({
+      #     local: { key: 'MASTER-KEY' }
```

Comment:
Is this example helpful? How can I best communicate that the master key must be a 96-byte base64-encoded string?

Reply:
Usually I tend to put this type of guidance/examples in the tutorial, where there is more flexibility with respect to expanding on a particular aspect of the driver.

Reply:
That's really helpful, thanks :)